pgmpy/readwrite/BIF.py

```diff
@@ -0,0 +1,312 @@
+        variable_states = {}                                                                # Creating an empty dictionary for variable states
+        variable_properties = {}                                                            # Creating an empty dictionary for variable prop   
+        word_expr = Word(alphanums+'_'+'-')                                                 # Defining a expression for valid word
+        name_expr = Suppress('variable')+ word_expr + Suppress('{')                         # Creating a expression for finding variable name
+        state_expr = ZeroOrMore( word_expr + Optional( Suppress(",") ) )
+
+        # Defining a variable state expression
+        variable_state_expr = Suppress('type') + Suppress(word_expr) + Suppress('[') + Suppress(Word(nums)) + \
+        Suppress(']') + Suppress('{') + Group(state_expr) + Suppress('}') + Suppress(';')
+        # variable states is of the form type description [args] { val1, val2 }; (comma may or may not be present)
+
+        property_expr = Suppress('property') + CharsNotIn(';') + Suppress(';')              # Creating a expr to find property
+
+        for block in variable_block:
+            temp = name_expr.searchString(block)                                            # Finding the string of the format name_expr
+            name = temp[0][0]                                                               # Assigning name the value of variable name
+            variable_names.append(name) 
+            temp = variable_state_expr.searchString(block)                                  # Finding the variable states
+            variable_states[name] = list(temp[0][0])                                        # Assigning the variable states in form of list
+            properties = property_expr.searchString(block)                                  # Getting the properties of variable                
+            variable_properties[name] = [x[0].strip() for x in properties ]
+
+        self.variable_names = variable_names
+        self.variable_states = variable_states
+        self.variable_properties = variable_properties
+        return
+
+    def get_variables(self):
+
+        """
+        Returns list of variables of the network
+
+        Sample run
+        -------------
+        >>> reader = BIF.BifReader("bif_test.bif")
+        >>> reader.get_variables()
+        ['light-on','bowel_problem','dog-out','hear-bark','family-out']
+        """
+        return self.variable_names
+
+    def get_states(self):
+
+        """
+        Returns the states of variables present in the network
+
+        Sample run
+        -----------
+        >>> reader = BIF.BifReader("bif_test.bif")
+        >>> reader.get_states()
+        {'bowel-problem': ['true','false'],
+         'dog-out': ['true','false'],
+         'family-out': ['true','false'],
+         'hear-bark': ['true','false'],
+         'light-on': ['true','false']}
+        """
+
+        return self.variable_states
```

Comment:
This would throw an error until the user has called `get_variable_info` before calling this method as you are assigning `self.variable_states` in that method. So add a check here for that.